runner/app/live/streamer/process.py

```diff
@@ -172,57 +172,86 @@ async def _initialize_pipeline(self):
                     raise
 
     async def _run_pipeline_loops(self):
-        pipeline = await self._initialize_pipeline()
-        input_task = asyncio.create_task(self._input_loop(pipeline))
-        output_task = asyncio.create_task(self._output_loop(pipeline))
-        param_task = asyncio.create_task(self._param_update_loop(pipeline))
+        self.pipeline = await self._initialize_pipeline()
+        input_task = asyncio.create_task(self._input_loop())
+        output_task = asyncio.create_task(self._output_loop())
+        param_task = asyncio.create_task(self._param_update_loop())
 
         async def wait_for_stop():
             while not self.is_done():
                 await asyncio.sleep(0.1)
-
-        tasks = [input_task, output_task, param_task, asyncio.create_task(wait_for_stop())]
+            logging.info("PipelineProcess: Pipeline loops interrupted, stopping pipeline")
 
         try:
-            await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
+            self.tasks = [input_task, output_task, param_task, asyncio.create_task(wait_for_stop())]
```

Comment:
Doesn't seem like this is used anywhere else, so it should be a regular variable and not a field. As stated in the other comment, we should generally avoid fields in this class as they can be misleading and error-prone.